[DOC-11870]: Feedback on Data Enrichment docs

RayOffiah opened the pull request:
### Description of Change

This pull request addresses feedback on the Data Enrichment documentation:
- Corrected a rogue occurrence of `without` to `with`.
- Made grammar and punctuation adjustments for better clarity and readability. 

### Type of Change

- [x] Documentation Update

### Checklist

- [ ] All new and existing tests pass.
- [x] Documentation has been updated where necessary.

----

### Preview

https://preview.docs-test.couchbase.com/docs-devex-DOC-11870-Feedback-on-Data-Enrichment--Couchbase-Docs/server/current/eventing/eventing-example-data-enrichment.html

RayOffiah committed:
[DOC-11870]: Feedback on Data Enrichment docs

Changed rogue occurrence of `without` to `with`.

Grammar and punctuation fixes.

Signed-off-by: Ray Offiah <ray.offiah@couchbase.com>

RichardSmedley commented on modules/eventing/pages/eventing-example-data-enrichment.adoc:
```diff
@@ -1,48 +1,66 @@
+In this example,
+if a new or updated document has two specifically named fields with IP addresses
+(representing the start and end of an address range),
+the Eventing Function uses `_get_numip_first_3_octets(ip)`
```
There's no leading underscore needed in get_numip_first_3_octets (afaict)

RayOffiah replied:
Got it. Not sure what I did there …

RichardSmedley reviewed (commented):
It's not my knowledge area, but apart from that undercore artefact it looks fine to me at a quick glance through - can approve if you need unblocking.

RayOffiah committed:
[DOC-11870]: Feedback on Data Enrichment docs

Changed rogue occurrence of `without` to `with`.

Grammar and punctuation fixes.

Signed-off-by: Ray Offiah <ray.offiah@couchbase.com>
